### DISCUSSION TITLE
Vector data type is Double,Java SDK only support FloatVector

### DISCUSSION THREAD

**Discussion by rilweic** [milvus-io/milvus#18094] [Q&A and General discussion]
my vector data like `0.07709378`, `-0.008429716` ,the type in java is Double  however Java api only support FloatVector.Is there any solution?

**Comment by yhmo** [comment]
Convert your double value to float.
The reason why milvus use float type for vector:
Most of the ANN search libraies(faiss, hnsw) are using float type for vector, float computation is faster than double, float type only requires half of memory/storage space compare to double.

**Reply by xiaofan-luan** [reply]
Let's keep as a issue, I believe support double vector is a reasonable requirement.

**Reply by PiercarloSlavazza** [reply]
@xiaofan-luan please consider also that OpenAI Ada emebeddings are returned as Double in recommended Java library - and obviously lots of people will start to use Ada emebddings…

**Reply by xiaofan-luan** [reply]
Good point.
I'll setup goal to support double embedding ASAP

**Reply by xiaofan-luan** [reply]
After discussion with team, I believe double, float16 and BF16 vector will be supported in 2.4 release.
Anyone interested, feel free to contribute

**Reply by brandonbiggs** [reply]
Is this still on track for 2.4 or did it make it into an earlier release?
